packages/openapi3/test/merge-patch.test.ts

```diff
@@ -0,0 +1,295 @@
+import { deepStrictEqual, ok } from "assert";
+import { describe, expect, it } from "vitest";
+import { OpenAPI3EmitterOptions } from "../src/lib.js";
+import { openApiFor } from "./test-host.js";
+
+export async function oapiForPatchRequest(
+  body: string,
+  modelDef: string,
+  contentType: string,
+  options: OpenAPI3EmitterOptions = {},
+) {
+  const oapi = await openApiFor(
+    `
+    @service(#{title: "Testing model"})
+    @route("/")
+    namespace root {
+      ${modelDef};
+      @patch op update(${body}): void;
+    }
+  `,
+    undefined,
+    options,
+  );
+
+  const content = oapi.paths["/"].patch.requestBody.content[contentType];
+  const useSchema = content.schema;
+
+  return {
+    isRef: !!useSchema.$ref,
+    schemas: oapi.components.schemas || {},
+    requestContent: content,
+  };
+}
+
+describe("merge-patch: MergePatch schema tests", () => {
+  it("emits correct schema for MergePatchUpdateReplaceOnly", async () => {
+    const req = await oapiForPatchRequest(
+      "@body body: Foo",
+      `
+      model Bar {
+        id: string;
+        description?: string;
+        @visibility(Lifecycle.Create)
+        createOnly?: string;
+        @visibility(Lifecycle.Update)
+        updateOnly?: string;
+        @visibility(Lifecycle.Read)
+        readOnly?: string;
+      }
+      
+      model Baz {
+        subject: Bar[];
+      }
+
+      model Foo is MergePatchUpdate<Baz>`,
+      "application/merge-patch+json",
+    );
+
+    ok(req.isRef);
+    ok(req.schemas.Foo, "expected definition named Foo");
+    ok(req.schemas.Foo.properties, "expected Foo to be an object");
+    ok(req.schemas.Foo.properties.subject, "expected Foo.subject to be a property");
```

Comment:
i would just skip thos assertions, if it fails with undefined below then we know something wasn't set and would make the test a bit cleaner

Reply:
It's fine with me to remove 60 and 61